dvc/repo/install.py

```diff
@@ -16,19 +16,19 @@ def pre_commit_install(scm: "Git") -> None:
     with modify_yaml(config_path) as config:
         entry = {
             "repo": "https://github.com/iterative/dvc",
-            "rev": "main",
+            "rev": "3.56.0",
```

Comment:
Let's not hardcode version. That will break our release workflow. 


```suggestion
            "rev": "".join(map(str, version_tuple[:3]),
```

You can import this from `from dvc import version_tuple`.

Reply:
Awesome, this is exactly what I was looking for. 

Out of curiosity, how does the versioning system work? 
I see that the package version is written to: 
https://github.com/iterative/dvc/blob/1e08cc5da3ded5ca6d67fc1941ceb46181d1ec8b/pyproject.toml#L159 
but this file is also ignored by `git`: 
https://github.com/iterative/dvc/blob/1e08cc5da3ded5ca6d67fc1941ceb46181d1ec8b/.gitignore#L46

So how does `version.py` retrieve the version number?
https://github.com/iterative/dvc/blob/1e08cc5da3ded5ca6d67fc1941ceb46181d1ec8b/dvc/version.py#L3

Reply:
As a side note, would it be feasible to automatically update the example `pre-commit-config.yaml` entry in the docs (https://github.com/iterative/dvc.org/pull/5323)? Or would just the static version be fine as an example?

Anecdotally, I stumbled upon `pre-commit` via the `dvc` docs, so some users might copy-paste the example `yaml` before knowing about or using the `dvc install --use-pre-commit-tool` command.

Reply:
We use https://github.com/pypa/setuptools-scm which generates `_dvc_version.py` for us when we build the package (or, when you do `pip install -e .` from a git repository or a git-archive).